Framework/Core/include/Framework/AnalysisDataModel.h

```diff
@@ -196,23 +198,32 @@ DECLARE_SOA_TABLE(UnassignedTracks, "AOD", "UNASSIGNEDTRACK",
 
 using UnassignedTrack = UnassignedTracks::iterator;
 
+namespace trigger
+{
+DECLARE_SOA_COLUMN(GlobalBC, globalBC, uint64_t);
+DECLARE_SOA_COLUMN(TriggerMask, triggerMask, uint64_t);
+} // namespace trigger
+
+DECLARE_SOA_TABLE(Triggers, "AOD", "TRIGGER", trigger::GlobalBC, trigger::TriggerMask);
+using Trigger = Triggers::iterator;
+
 namespace calo
 {
-DECLARE_SOA_INDEX_COLUMN(Collision, collision);
+DECLARE_SOA_INDEX_COLUMN_FULL(GlobalBC, globalBC, uint64_t, Triggers, "fGlobalBC");
```

Comment:
This is not nice here because it creates a column called GlobalBCId but GlobalBC is already an id in itself. Please make suggestions on the naming

Reply:
The index columns are declared as referring to a table type, here you declare an index to a "GlobalBC" table that is not defined. The column then is filled with row numbers of the indexed table - which is how the access by index would work. Otherwise you will have to add a column with actual BCs to calo (and other tables).

Additionally, if the GlobalBC table is generated-on-the fly, then the index columns of all the dependent tables need to be _refilled_ with corresponding row values,

Reply:
I think the 4th parameter is the table name, not the first? Check the usage for PosTrack in the data model...

Reply:
@jgrosseo sorry! Did not scroll to the right enough, the Triggers table is still there. However, will there be trigger info always attached to every BC? Similar to other data, maybe Triggers should index  a bigger BC table instead?

Reply:
That would be an option, but I wonder if a table with just one additional column is worth it. We could put TriggerMask to NULL in case we don't have info for that particular BC. Most BC will have a TriggerMask. What do you think?

Reply:
The advantage will be that Trigger table would be contiguous, meaning we can avoid checking for NULL when iterating. This is, of course, about memory layout.

Reply:
In practice we should have trigger info in every bc. Even if CTP is not sending data for some reason, we can always fill trigger info in missing bcs with dummy values.